Rewritten into java

sye8 opened the pull request:
Rewritten the filter into java for java application

sye8 committed:
Rewritten into java

wouterbulten commented on src/KalmanFilter.java:
```diff
@@ -0,0 +1,77 @@
+ * @author yesifan
+ * 
+ * @see https://github.com/wouterbulten/kalmanjs
+ *
```
Can you add the license of the project here? See for example the docblock on the js file.

wouterbulten commented on src/KalmanFilter.java:
```diff
@@ -0,0 +1,77 @@
+package sye8.utils;
```
Could you rename this package?

wouterbulten commented on src/KalmanFilter.java:
```diff
@@ -0,0 +1,77 @@
+		return this.x;
+	}
+
+	public void setMeasurementNoise(double noise){
```
Please add the docblock here

wouterbulten commented on src/KalmanFilter.java:
```diff
@@ -0,0 +1,77 @@
+		    this.Q = noise;
+	}
+	
+	public void setProcessNoise(double noise){
```
Please add the docblock here

wouterbulten reviewed (changes requested):
Hi @ysf199711! Thank you for your pull request. This repo was originally only for the Javascript version of the 1D kalman filter (hence the JS in the name).

However, I get a lot of questions regarding a Java implementation so there is merit in this translation. I would like, however, to keep the main repo/dir clean of any non-js files. Instead, I would like to add a `contrib` folder with contributions.

So, could you move this file to: /contrib/java/KalmanFilter.java? I also added some comments to your code.

After the changes I will merge the branch. Again, thanks for your contributions!

sye8 committed:
Change location and update comments

Package name undecided yet. Original package name was to be used in another project.

sye8 commented:
Location and comments updated. Package name will be left as (default) as new package name is undecided. Added LGPL v3 license

sye8 committed:
Create LICENSE

sye8 committed:
Update location

wouterbulten commented:
Thanks for the changes! I will merge it in the next few days
